avocado/core/mux.py

```diff
@@ -163,15 +164,25 @@ def initialize_mux(self, root, mux_path, debug):
         self.root = root
         self.mux_path = mux_path
         self.debug = debug
+        self.variant_ids = self._get_variant_ids()
+
+    def _get_variant_ids(self):
+        variant_ids = []
+        for variant in MuxTree(self.root):
+            variant.sort(key=lambda x: x.path)
+            fingerprint = "-".join(_.fingerprint() for _ in variant)
+            variant_ids.append("-".join(node.name for node in variant) + '-' +
```

Comment:
I'm not comparing it to the indexes. I'm comparing it to a truncated version of itself. But yes, maybe we should only worry about that when it become an issue.